shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -218,6 +222,16 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
   flutterArguments.command_line_argc = static_cast<int>(argv.size());
   flutterArguments.command_line_argv = argv.size() > 0 ? argv.data() : nullptr;
   flutterArguments.platform_message_callback = (FlutterPlatformMessageCallback)OnPlatformMessage;
+  flutterArguments.update_semantics_node_callback = [](const FlutterSemanticsNode* node,
+                                                       void* user_data) {
+    FlutterEngine* engine = (__bridge FlutterEngine*)user_data;
+    [engine updateSemanticsNode:node];
```

Comment:
I thought we got rid of the user_data part and just reworked it so we could construct it the way we'd expect it ot be constructed.

I know we talked about that anyway. If it didn't land that way can we update it now?

Reply:
Is it possible? I remember i have tried doing it without user data, but I keep getting into issue that the callback here cannot capture local reference.

Reply:
You should be able to lambda capture it

Reply:
update_semantics_node_callback and update_semantics_custom_action_callback are function type. I took a look at https://stackoverflow.com/questions/11406678/lambda-capture-causes-incompatible-operand-types-error. It seems like I cannot assign capture lambda to it?